quickjs.c

```diff
@@ -474,8 +474,8 @@ struct JSString {
     struct list_head link; /* string list */
 #endif
     union {
-        uint8_t str8[0]; /* 8 bit strings will get an extra null terminator */
-        uint16_t str16[0];
+        __extension__ uint8_t str8[0]; /* 8 bit strings will get an extra null terminator */
+        __extension__ uint16_t str16[0];
```

Comment:
I suppose this needs to be flagged as an `__extension__` and use the gnu zero-length syntax because unions cannot contain flexible arrays, let alone multiple flexible arrays?

Reply:
Yep exactly